src/utils/polygonUtils.cpp

```diff
@@ -1255,6 +1255,10 @@ bool PolygonUtils::polygonCollidesWithLineSegment(const Polygons& polys, const P
 
 bool PolygonUtils::polygonCollidesWithLineSegment(const Polygons& polys, const Point& startPoint, const Point& endPoint)
 {
+    if(endPoint == startPoint)
+    {
+        return false; //Zero-length line segments never collide.
```

Comment:
Not even when equal to one of the polygons' points?

Reply:
That's an edge case. It would be up to the user of the function whether that would need to trigger a collision.

In the `PathOrderOptimizer` and the `Comb` classes it is not desired, because a zero-length line segment would then trigger a combing calculation which finds the intersections between the line segment and a polygon. To find the intersections, it'll try to rotate the polygon and line segment such that the line segment is aligned with the X axis. Since the line segment has zero length, this calculation yields a transformation matrix which is broken (it includes some `infinity()` floats).

The only other place where this is used is in the infill calculations to see when to break up infill lines and to join them. Zero-length infill lines aren't really relevant there since they don't get printed.

So thus far the other interpretation of the edge case is not necessary anywhere.

Reply:
>  It would be up to the user of the function whether that would need to trigger a collision.

Document in header?